kaldb/src/main/java/com/slack/kaldb/logstore/search/aggregations/HistogramAggBuilder.java

```diff
@@ -0,0 +1,90 @@
+package com.slack.kaldb.logstore.search.aggregations;
+
+import java.util.List;
+import java.util.Map;
+
+/** Aggregation request type to form a histogram bucketed by a timestamp */
+public class HistogramAggBuilder extends ValueSourceAggBuilder {
+  public static final String TYPE = "histogram";
+  private final String interval;
+  private final long minDocCount;
+
+  public HistogramAggBuilder(String name, String fieldName, String interval) {
+    super(name, fieldName);
+    this.interval = interval;
+    this.minDocCount = 1;
+  }
+
+  public HistogramAggBuilder(
+      String name,
+      String fieldName,
+      String interval,
+      long minDocCount,
+      List<AggBuilder> subAggregations) {
+    // todo - metadata?
+    super(name, Map.of(), subAggregations, fieldName);
+
+    this.interval = interval;
+    this.minDocCount = minDocCount;
+  }
+
+  public String getInterval() {
+    return interval;
+  }
+
+  public double getIntervalDouble() {
```

Comment:
Maybe a comment somewhere why you chose String as the type, and then provided a double? Just a suggestion, not required.